File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/CustomerInformationEndpointDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/customers")
public class CustomerController {

    @GetMapping("/{customerId}")
    public ResponseEntity<Map<String, Object>> getCustomer(
            @PathVariable Long customerId) {

        // Sample customer data
        Map<String, Object> customer = new LinkedHashMap<>();
        customer.put("customerId", customerId);
        customer.put("fullName", "John Doe");
        customer.put("email", "john.doe@example.com");
        customer.put("phone", "+1-555-123-4567");
        customer.put("status", "Active");

        // JSON response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("message", "Customer retrieved successfully.");
        response.put("customer", customer);

        return ResponseEntity.ok(response);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/customer_information_endpoint_detailed-chatgpt.py
from flask import Flask, jsonify

app = Flask(__name__)

@app.route("/customers/<int:customer_id>", methods=["GET"])
def get_customer(customer_id):
    # Simulated customer data
    customer = {
        "customerId": customer_id,
        "firstName": "John",
        "lastName": "Doe",
        "email": "john.doe@example.com",
        "phone": "+1-555-123-4567",
        "status": "Active"
    }

    # Format the returned data
    formatted_customer = {
        "id": customer["customerId"],
        "fullName": f"{customer['firstName']} {customer['lastName']}",
        "email": customer["email"],
        "phone": customer["phone"],
        "status": customer["status"]
    }

    # JSON response
    response = {
        "success": True,
        "message": "Customer retrieved successfully.",
        "customer": formatted_customer
    }

    return jsonify(response), 200

if __name__ == "__main__":
    app.run(debug=True)